uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlementTest.php

```diff
@@ -1232,6 +1233,76 @@ public function testGetBreakdownWillGroupTogetherOverriddenBalanceChangeWithLeav
     $this->assertEquals($expectedResult, $result);
   }
 
+  public function testGetForLeaveRequestReturnsTheLeavePeriodEntitlementForAGivenLeaveRequest() {
+    $absencePeriod1 = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('-10 days'),
+      'end_date' => CRM_Utils_Date::processDate('-1 day'),
+    ]);
+
+    $absencePeriod2 = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('today'),
+      'end_date' => CRM_Utils_Date::processDate('+10 days'),
+    ]);
+
+    $periodEntitlement1 = LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => 1,
+      'period_id' => $absencePeriod1->id,
+      'type_id' =>  1
+    ]);
+
+    $periodEntitlement2 = LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => 1,
+      'period_id' => $absencePeriod2->id,
+      'type_id' =>  1
+    ]);
+
+    $periodEntitlement3 = LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => 1,
+      'period_id' => $absencePeriod2->id,
+      'type_id' =>  2
+    ]);
+
+    $leaveRequest1 = LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $periodEntitlement1->contact_id,
+      'type_id' => $periodEntitlement1->type_id,
+      'from_date' => CRM_Utils_Date::processDate('-5 days'),
+      'to_date' => CRM_Utils_Date::processDate('-3 days'),
+    ]);
+
+    $leaveRequest2 = LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $periodEntitlement2->contact_id,
+      'type_id' => $periodEntitlement2->type_id,
+      'from_date' => CRM_Utils_Date::processDate('+2 days'),
+      'to_date' => CRM_Utils_Date::processDate('+2 days'),
+    ]);
+
+    $leaveRequest3 = LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $periodEntitlement3->contact_id,
+      'type_id' => $periodEntitlement3->type_id,
+      'from_date' => CRM_Utils_Date::processDate('+9 days'),
+      'to_date' => CRM_Utils_Date::processDate('+10 days'),
+    ]);
+
+    $this->assertEquals($periodEntitlement1->id, LeavePeriodEntitlement::getForLeaveRequest($leaveRequest1)->id);
+    $this->assertEquals($periodEntitlement2->id, LeavePeriodEntitlement::getForLeaveRequest($leaveRequest2)->id);
+    $this->assertEquals($periodEntitlement3->id, LeavePeriodEntitlement::getForLeaveRequest($leaveRequest3)->id);
+  }
+
+  /**
+   * @expectedException RuntimeException
+   * @expectedExceptionMessage It was not possible to find an AbsencePeriod containing the given LeaveRequest
+   */
+  public function testGetForLeaveRequestShouldThrowAnExceptionIfThereIsNoAbsencePeriodContainingTheGivenLeaveRequest() {
```

Comment:
We can change the test name to `testGetForLeaveRequestShouldThrowAnExceptionIfThereIsNoAbsencePeriodContainingTheGivenLeaveRequestDates` to be more descriptive